contrib/babelfishpg_tsql/src/pltsql_coerce.c

```diff
@@ -885,6 +917,243 @@ run_tsql_best_match_heuristics(int nargs, Oid *input_typeids, FuncCandidateList
 	return new_candidates;
 }
 
+/*
+ * getImmediateBaseTypeOfUDT()
+ * This funciton 
+ */
+static Oid getImmediateBaseTypeOfUDT(Oid typeid)
+{
+	Relation					relsetting;
+	HeapTuple					tuple;
+	ScanKeyData					scankey;
+	SysScanDesc					scan;
+	bool						isnull;
+	Datum						datum;
+	Datum                       tsql_typename;
+	LOCAL_FCINFO(fcinfo, 1);
+
+	/* if common_utility_plugin_ptr is not initialised */
+	if (common_utility_plugin_ptr == NULL)
+		ereport(ERROR,
+				(errcode(ERRCODE_INTERNAL_ERROR),
+					errmsg("Failed to find common utility plugin.")));
+
+	/* if tsql_typename is NULL it implies that inputTypId corresponds to UDT */
+	InitFunctionCallInfoData(*fcinfo, NULL, 0, InvalidOid, NULL, NULL);
+	fcinfo->args[0].value = ObjectIdGetDatum(typeid);
+	fcinfo->args[0].isnull = false;
+	tsql_typename = (*common_utility_plugin_ptr->translate_pg_type_to_tsql) (fcinfo);
+
+	/* if given type is not an UDT then return InvalidOid */
+	if (tsql_typename)
+		return InvalidOid;
+
+	/* Get immediate base type id of given type id */
+	relsetting = table_open(TypeRelationId, AccessShareLock);
```

Comment:
got a suggestion to use `heap_getattr` [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/pull/2495#discussion_r1575706878) and since `heap_getattr` function requires relsetting for it's third parameter I used this approach than using `SearchSysCache1`.
cc: @Deepesh125

Reply:
With `SearchSysCache1`, you don't need to use `heap_getattr`. Please check PG code for reference.

Reply:
You can use `SysCacheGetAttr` along with `SearchSysCache1` to get the atrribute from tuple.